Add rake task to edit a document's base_bath

AlanGabbianelli opened the pull request:
Add rake task to edit a document's base_bath

Trello card:
https://trello.com/c/kTKWF48j/530-document-how-to-change-a-url-in-specialist-publisher

How to change a URL in Specialist Publisher wasn't a documented process.
Making it into a rake task seems a nicer solution than simply
documenting how to do it and then having to do it manually.

Now, in order to edit a document's base_path, it's possible to just use
the rake task passing the document's `content_id` and the new
`base_path` as arguments.

Commit:
Add rake task to edit a document's base_bath

Trello card:
https://trello.com/c/kTKWF48j/530-document-how-to-change-a-url-in-specialist-publisher

How to change a URL in Specialist Publisher wasn't a documented process.
Making it into a rake task seems a nicer solution than simply
documenting how to do it and then having to do it manually.

Now, in order to edit a document's base_path, it's possible to just use
the rake task passing the document's `content_id` and the new
`base_path` as arguments.

barrucadu commented on lib/tasks/base_path.rake:
```diff
@@ -0,0 +1,24 @@
+    begin
+      document = Document.find(content_id)
+    rescue DocumentFinder::RecordNotFound => e
+      puts "Error finding the document: #{e.inspect}"
```
It looks like there are actually a few different exceptions which could be raised here:

```
irb(main):001:0> document = Document.find('')
...
GdsApi::TimedOutException (GdsApi::TimedOutException)

irb(main):003:0> document = Document.find('foo')
...
DocumentFinder::RecordNotFound (Document: foo)

irb(main):004:0> document = Document.find('foo bar baz')
...
GdsApi::InvalidUrl (GdsApi::InvalidUrl)
```

I'm not sure that catching the exception and displaying a message is any more useful than just letting the exception give a stack trace and terminate the task.

AlanGabbianelli replied:
Yes, I wasn't sure about this either, to be honest.
I followed the pattern that I saw already implemented in the code, but I'm usually a fan of letting things fail and dealing with the consequences.
I'll change it.

barrucadu commented on lib/tasks/base_path.rake:
```diff
@@ -0,0 +1,24 @@
+      puts "Error finding the document: #{e.inspect}"
+    end
+
+    if document
```
If the exception were allowed to propagate, above, then this check wouldn't be needed.

barrucadu commented on lib/tasks/base_path.rake:
```diff
@@ -0,0 +1,24 @@
+        puts "The #{document.class} with title \"#{document.title}\" has been successfully edited"
+        puts "Its new base_path is now \"#{document.base_path}\""
+      else
+        puts "Couldn't edit the #{document.class} with title \"#{document.title}\""
```
If saving fails, there will probably be an associated validation error.  Perhaps use `document.save!`, which throws an exception if there's something wrong.

bevanloon replied:
don't think there's a `save!` method here - specialist publisher ~~isn't a~~ doesn't use it's own local database ~~backed application~~ so this isn't a traditional ActiveRecord model.

barrucadu commented on lib/tasks/base_path.rake:
```diff
@@ -0,0 +1,24 @@
+    if document
+      document.base_path = base_path
+      document.update_type = 'minor'
+      if document.save
```
Before making the change it can be nice to print out what it will be.  Just something simple like

```
puts "#{document.base_path} -> #{base_path}"
```

Commit:
Remove rescue and let it fail

Letting it fail, we get a useful stack trace that will point us to the 
right direction.
